File: Easy/Python/1005. Maximize Sum Of Array After K Negations.py
class Solution(object):
    def largestSumAfterKNegations(self, nums, k):
        nums.sort()
        n = len(nums)

        for i in range(n):
            if k != 0 and nums[i] < 0:
                nums[i] *= -1
                k -= 1
            else:
                break
        
        if k != 0:
            nums.sort()
            while k != 0:
                nums[0] *= -1
                k -= 1
        
        this_sum = 0
        for i in range(n):
            this_sum += nums[i]
        
        return this_sum

File: Easy/JavaScript/1005. Maximize Sum Of Array After K Negations.js
var largestSumAfterKNegations = function(nums, k) {
    nums.sort((a, b) => a - b);
    let n = nums.length;

    for(let i = 0; i < n; i++){
        if(k != 0 && nums[i] < 0){
            nums[i] *= -1;
            k--;
        }else{
            break;
        }
    }

    if(k != 0){
        nums.sort((a, b) => a - b);
        while(k != 0){
            nums[0] *= -1;
            k--;
        }
    }

    let sum = 0;
    for(let i = 0; i < n; i++){
        sum += nums[i];
    }

    return sum;
};
